lib/location/method_gnss.c

```diff
@@ -226,6 +235,14 @@ static void method_gnss_agps_request_work_fn(struct k_work *item)
 		.fragment_size = 0
 	};
 
+	err = location_utils_check_modem_supports_lte();
+	if (err) {
+		LOG_ERR("nRF Cloud A-GPS request failed: %s",
+			(err == -EINVAL) ? "Current modem system mode does not support LTE"    :
+					   "Current modem system mode could not be determined");
```

Comment:
This is nitpicking but instead of copying these strings into few places, I'd think of logging this in location utils functions. There is already a trace for `-EBUSY` case but not for `-EINVAL`. If this is left as it is, I hope compiler optimizes these strings as possible.

Reply:
> And, in retrospect, I tend to agree with @tokangas on this one, it makes the functions in `location_utils` cleaner (their sole purpose is to detect modem feature support, rather than serve dual purpose by detecting and alerting to problems. 

Ideally yes. However, in practice the most accurate information is often within the function that first observe that error occurred so writing an error message there kind of makes sense. From design perspective, what would happen if those util functions were changed and there would be 3rd and maybe 4th return value. Or if there would be one or two more places calling these functions.

> Personally, I think the benefits of the more explicit, easier to interpret error message is outweighed by the cost of ~300 bytes of flash.

I don't see why you couldn't have the same error message content in the utils functions in this case. And while 200-300 bytes is quite little, we are using it here a little unnecessarily.

> Perhaps for this same reason, I should remove the error messages on the `-EBUSY` cases?).

If we keep it the way it's implemented now, I think you should do this.

But let's see what @tokangas says. I guess two Tommi's need to have a chat :) But let's not make this too big of a deal as in this specific case, it really doesn't matter that much which way this is done.

Reply:
> The util functions have no knowledge of whether they are being called by AGPS or by PGPS functions, so they cannot indicate which of the two failed. We could make a generic message mentioning both, but then if any other part of location library decides to use the util functions, the error message would no longer be accurate. Passing in the calling reason as an argument is another option, but that would make the code hard to follow, IMO

Yep, but I'm assuming you have two logs as you wrote earlier: `"A-GPS request failed" and "Error: current modem system mode does not support LTE"`.

I don't see a big problem with this given it's an error and it can raise attention for someone who is looking at the logs. It's a log anyway and not the real output of an end application.

Reply:
Let's leave it as it is. And let's not add another error trace in the utils functions because it may not be an error condition for the caller.

Reply:
I like the current approach. If you think from `location_utils_check_modem_supports_gnss()` point of view, it logs an error if checking for GNSS support fails. If support can be queried from modem, but GNSS is not supported, it's not considered an error. The check was successful and GNSS is not supported.

From `method_gnss_location_get()` point of view, it is considered an error if checking for GNSS support fails or GNSS is not supported. But a different function might also call `location_utils_check_modem_supports_gnss()`, but not consider missing GNSS support an error.

Most likely the compiler optimizes the duplicate strings out, as long as the strings are identical. And these are only traces, so it's also possible to disable them using Kconfig.